### PR TITLE
Make sure dynamo give us different graph when input shape changes

### PR DESCRIPTION
**Pull request by JackCaoG**
Add a test to make sure when DS is enabled on dynamo it does not break the assumption that XLA has on dynamo bridge.

FYI @ezyang

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: test/dynamo/test_dynamo.py [hunk 1]
```diff
@@ -68,6 +68,25 @@ def test_simple_model(self):
...
     self.assertTrue(torch.allclose(res_cpu_3, res_xla_dynamo_3.cpu()))
 
+  def test_simple_model_with_different_input_shape(self):
+    met.clear_counters()
+    device = xm.xla_device()
+    xla_x = torch.randn(5, 5).to(device)
+    xla_y = torch.randn(5, 5).to(device)
+    xla_z = torch.randn(10, 10).to(device)
+    self.fn_simple_dynamo(xla_x, xla_x)
+    compiel_count = met.metric_data('CompileTime')[0]
```
Comment (wonjoolee95) [inline test/dynamo/test_dynamo.py:78]: nit: `compile` spelling 😄

**Review by wonjoolee95 — APPROVED** [review]

File: test/dynamo/test_dynamo.py (hunk 1 above)
Comment (JackCaoG) [reply test/dynamo/test_dynamo.py]: nice catch!

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Comment by alanwaketan** [comment]
Not sure how this is related to DS though...

**Comment by JackCaoG** [comment]
Ed was talking about enabling the DS on dynamo directly. I am worried once that happened, dynamo will start thinking about
```
resnet([18,3,512])
```
and
```
resnet([8,3,512])
```
to be the same program. Pretty much I am not sure if dynamo will guard on input shape anymore, but we need it.

**Comment by alanwaketan** [comment]
If Dynamo won't guard on input shapes after DS is enabled, it will blow out my mind. Nevertheless, I appreciate the additional guards.

**Pull request merged**